### PR TITLE
machine/samd21: extract shared functionality for processor family

### PR DESCRIPTION
**Pull request by deadprogram**
This PR extracts the functionality for the ATSAMD21 processor into files that can be shared for any boards based on this family.

### CODE REVIEW

File: targets/atsamd21.json [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
 	"cflags": [
 		"--target=armv6m-none-eabi",
 		"-Qunused-arguments"
 	],
 	"ldflags": [
-		"-T", "targets/atsamd21g18.ld"
+		"-T", "targets/atsamd21.ld"
 	],
 	"extra-files": [
 		"src/device/sam/atsamd21g18a.s"
```
Comment (aykevl) [inline targets/atsamd21.json]: This option depends on the specific chip, and looking at `src/device/sam/atsamd21*.s` there are small differences in supported peripherals. I think it should be moved into the `itsybitsy-m0.json` file, or perhaps in a new `atsamd21g18.json` file.

**Comment by aykevl** [comment]
Here is a SAMD21 chip that has just 4kB of RAM and 32kB of flash: https://www.digikey.com/product-detail/en/microchip-technology/ATSAMD21G15B-MUT/ATSAMD21G15B-MUTTR-ND/5325446

**Comment by deadprogram** [comment]
OK renamed and made somewhat less generalized.

**Comment by aykevl** [comment]
Merged in 942d4903ce4ad5a4257eed9ba1e7dbee744e18c3.

**Pull request closed without merging**